NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp

```diff
@@ -150,14 +151,35 @@ void key_callback(GLFWwindow *window, int key, int scancode, int action, int mod
    if (glfwGetKey(window, GLFW_KEY_ESCAPE) == GLFW_PRESS)
       glfwSetWindowShouldClose(window, true);
 
-   if (glfwGetKey(window, GLFW_KEY_W) == GLFW_PRESS)
-      camera.ProcessKeyboard(Camera::FORWARD, deltaTime);
-   if (glfwGetKey(window, GLFW_KEY_S) == GLFW_PRESS)
-      camera.ProcessKeyboard(Camera::BACKWARD, deltaTime);
-   if (glfwGetKey(window, GLFW_KEY_A) == GLFW_PRESS)
-      camera.ProcessKeyboard(Camera::LEFT, deltaTime);
-   if (glfwGetKey(window, GLFW_KEY_D) == GLFW_PRESS)
-      camera.ProcessKeyboard(Camera::RIGHT, deltaTime);
+   if (key == GLFW_KEY_W && action == GLFW_PRESS)
+      camera.goForward = true;
+   if (key == GLFW_KEY_W && action == GLFW_RELEASE)
```

Comment:
I feel like these just being `if(...)` could be improved... thou it allows multiple movements at the same time =? 
```c++
if (key == GLFW_KEY_W && action == GLFW_PRESS) camera.goForward = true; else camera.goForward = false;
```

Reply:
Writing it as an if else statement does not work. The reason I did it this way is because I specify that I want the key to be released for the movement to stop.
Nevertheless, I tried implementing using if else just to see how it works, and I was right, doesn't move smoothly.